modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/TokenCountFieldMapper.java

```diff
@@ -202,8 +211,11 @@ protected String contentType() {
 
     @Override
     protected void mergeOptions(FieldMapper other, List<String> conflicts) {
-        this.analyzer = ((TokenCountFieldMapper) other).analyzer;
-        this.enablePositionIncrements = ((TokenCountFieldMapper) other).enablePositionIncrements;
+        // TODO we should ban updating analyzers and null values as well
+        if (this.enablePositionIncrements != ((TokenCountFieldMapper)other).enablePositionIncrements) {
+            conflicts.add("mapper [" + name() + "] has a different [enable_position_increments] setting");
```

Comment:
this seems like a new error that was not returned before?

Reply:
You're right.  I think this is the right choice (if you've indexed some docs while disabling increments, and then you index further docs after they have been enabled, the numbers stored in the two docs won't be comparable), but it should be a warning in 7x instead of an error.